src/select2.component.ts

```diff
@@ -148,6 +148,29 @@ export class Select2Component implements AfterViewInit, OnChanges, OnDestroy, On
             this.renderer.setElementProperty(this.selector.nativeElement, 'disabled', this.disabled);
         }
     }
+     
+    private setElementValue ( newValue ) {
+        // NOTE:
+        // Multiple values are received from ngOnChanges as a string with single values separated by comma.
+        // Select2OptionData id values should not contain commas for predictable results.
+        // While using commas in ids is a bad idea generally, here may cause malfunctions.
+
+        if (newValue.indexOf(',') != -1) {
```

Comment:
I think this is not a correct way to do it, what if user provide single value that contains `,`? I think that we need to check if value that is provided is array and then use it as multi value, otherwise always use single value as string.